dynamiq/nodes/llms/openai.py

```diff
@@ -21,3 +28,88 @@ def __init__(self, **kwargs):
         if kwargs.get("client") is None and kwargs.get("connection") is None:
             kwargs["connection"] = OpenAIConnection()
         super().__init__(**kwargs)
+
+    def is_o_series_model(self) -> bool:
```

Comment:
It seems this could be `@cached_property` as we don't need to recheck each time as we don't update the model dynamically.